tests/storage/zfsvol/conftest.py

```diff
@@ -0,0 +1,35 @@
+import logging
+import pytest
+
+# Explicitly import package-scoped fixtures (see explanation in pkgfixtures.py)
+from pkgfixtures import host_with_saved_yum_state, sr_disk_wiped
+
+@pytest.fixture(scope='package')
+def host_with_zfsvol(host_with_saved_yum_state):
+    host = host_with_saved_yum_state
+    host.yum_install(['zfs']) # FIXME xcp-ng-xapi-storage-zfs-ng ?
+    host.ssh(['modprobe', 'zfs'])
+    yield host
+
+@pytest.fixture(scope='package')
+def zfsvol_sr(host, sr_disk_wiped, host_with_zfsvol):
+    """ A ZFS Volume SR on first host. """
+    sr = host.sr_create('zfs-vol', "ZFS-local-SR-test", {'device': '/dev/' + sr_disk_wiped})
+    yield sr
+    # teardown violently - we don't want to require manual recovery when a test fails
+    sr.forget()
```

Comment:
I think that it won't work bcause unplug PBDs will probably failed. At least most of the time I tried it fails because PBD cannot be unplugged if there are some dataset (volume). But ok it may work  sometimes but when test fails I think that we will end with SR not forget. But ok it is better than nothing maybe ;)

Reply:
sr-forget has no problem with remaining VDI (as long as there is eg. no running VM using them), as opposed to sr-destroy, this is why it works much better in case of test failure than trying to be symmetric and calling `sr.destroy()`.

Reply:
I just tested a forget with a remaining VDI (no running VM using them because no running VM created) and I got:
```
[10:10 xcp-ng-opkwktak ~]# xe sr-forget uuid=fadda3f0-834a-50db-adce-b64520d2e82b
The SR is still connected to a host via a PBD. It cannot be destroyed or forgotten.
sr: fadda3f0-834a-50db-adce-b64520d2e82b (ZFS-VOL SR)
```

Reply:
Note for @stormi if the sr.forget is a success then PBDs are deleted and don't exist anymore.

Reply:
But ok I just look the code of "forget" in sr.py and it is doing the unplug of the PBDs so it should work :) ... and PBD are deleted after being unplugged and sr forgotten...

Reply:
> The SR is still connected to a host via a PBD

`xe pbd-unplug` must be performed before any sr-forget or sr-destroy

Reply:
and it is the same for VDIs at least if they are not attached to a VM as @ydirson said earlier.

Reply:
> > The SR is still connected to a host via a PBD
> 
> `xe pbd-unplug` must be performed before any sr-forget or sr-destroy

Yes and it is why I added that in fact `sr.forget()` is doing the unplugged (it is not clear just regarding the name of the function that suggest only a call to sr.forget()). So yes with the unplugged it should be ok.

Reply:
Indeed this code was duly exercised by the various tests written before the code implements the features.